src/dimension_slice.c

```diff
@@ -744,13 +744,14 @@ dimension_slice_check_chunk_stats_tuple_found(TupleInfo *ti, void *data)
 
 	foreach (lc, chunk_ids)
 	{
-		BgwPolicyChunkStats *chunk_stat =
-			ts_bgw_policy_chunk_stats_find(info->job_id, lfirst_int(lc));
+		int chunk_id = lfirst_int(lc);
+		BgwPolicyChunkStats *chunk_stat = ts_bgw_policy_chunk_stats_find(info->job_id, chunk_id);
 
```

Comment:
Could you add a comment here
```suggestion
Look for a chunk that a) doesn't have a job stat (reorder ) and b) is not compressed (should not reorder compressed chunk)
```